py/server/deephaven_internal/auto_completer/_signature_help.py

```diff
@@ -0,0 +1,245 @@
+            description += "\n\n"
+
+    if len(docs.examples) > 0:
+        description += "#### **Examples**\n\n"
+        for example in docs.examples:
+            if example.description is not None and example.description.startswith(">>> "):
+                description += f"```\n{example.description}\n```"
+            else:
+                description += example.description
+            description += "\n\n"
+
+    return description.strip()
+
+
+def _generate_display_sig(signature: Signature) -> str:
+    """
+    Generate the signature text for the signature help. Truncates the signature if it is too long. If the current
+    argument is positional, it will display the next 2 arguments. If the current argument is keyword, it will only
+    display the current argument.
+    """
+
+    if len(signature.to_string()) <= MAX_DISPLAY_SIG_LEN:
+        return signature.to_string()
+    
+    # Use 0 as default to display start of signature
+    index = signature.index if signature.index is not None else 0
+    display_sig = f"{signature.name}("
+
+    if index > 0:
+        display_sig += "..., "
+
+    # If current arg is positional, display next 2 args
+    # If current arg is keyword, only display current args
+    if signature.params[index].kind in POSITIONAL_KINDS:
+        # Clamp index so that 3 args are shown, even at last index
+        index = max(min(index, len(signature.params) - 3), 0)
+        end_index = index + 3
+        # If the next arg is not positional, do not show the one after it
+        # Otherwise, this arg will show 2 ahead, and then next arg will show 0 ahead
+        if signature.params[index + 1].kind not in POSITIONAL_KINDS:
+            end_index -= 1
+        display_sig += ", ".join([param.to_string() for param in signature.params[index:end_index]])
+        if index + 3 < len(signature.params):
+            display_sig += ", ..."
+    else:
+        display_sig += signature.params[index].to_string()
+        if index + 1 < len(signature.params):
+            display_sig += ", ..."
+
+    return display_sig + ")"
+
+
+def _generate_param_markdowns(signature: Signature, params: list[Any]) -> list[Any]:
+    """
+    Generate markdown for each parameter in the signature. This will be shown on top of the description markdown.
+    """
+
+    param_docs = []
+    for i in range(len(signature.params)):
+        if signature.params[i].to_string().strip() in IGNORE_PARAM_NAMES:
+            continue
+
+        param = params[i]
+        description = f"##### **{param['name']}**"
+        if param['type'] is not None:
+            description += f" : *{param['type']}*"
+        description += "\n\n"
+        if param['description'] is not None:
+            description += f"{param['description']}\n\n"
+        description += "---"
+
+        param_docs.append([signature.params[i].to_string().strip(), description])
+
+    return param_docs
+
+
+def _get_signature_help(signature: Signature) -> list[Any]:
```

Comment:
Should this have the `_` prefix removed since it is the method from the file that is used elsewhere?